app/src/main/java/org/gem/indo/dooit/controllers/goal/GoalAddController.java

```diff
@@ -154,8 +154,16 @@ private void doPopulate(Map<String, Answer> answerLog) {
             goal.setImageFromProto(true);
         }
 
-        // Goal is stored in case the view is destroyed during the conversation
-        persisted.saveConvoGoal(botType, goal);
+        try {
+            // Goal is stored in case the view is destroyed during the conversation
+            persisted.saveConvoGoal(botType, goal);
+        }catch (IllegalArgumentException e){
+            CrashlyticsHelper.log(TAG, "do Populate (addGoal): ", "goal start date: " + goal.getStartDate() +
+                    " Target amount: " + goal.getTarget() + " Goal name: " + goal.getName() +
+            " Goal Weekly Target: " + goal.getWeeklyTarget());
+
+            CrashlyticsHelper.log(TAG, "do Populate (addGoal)( persisted.saveConvoGoal(botType, goal)): ", e.getMessage());
```

Comment:
Use `CrashlyticsHelper.logException(Exception e)` to send a non-fatal exception to Crashlytics.